src/core/Makefile.am

```diff
@@ -81,6 +81,7 @@ libopenthread_a_SOURCES             = \
     thread/panid_query_server.cpp     \
     thread/thread_netif.cpp           \
     thread/thread_tlvs.cpp            \
+    utils/global_address.cpp          \
```

Comment:
Take a look at this commit for an example of adding new files:
https://github.com/openthread/openthread/commit/20717169363c3aa600f1149225368538361b901b